### PR TITLE
COMPONENT: Toggle Button

### PR DESCRIPTION
**Pull request by Nap1400**
image

### CODE REVIEW

**Comment by raynesmike** [comment]
dont forget to fix the conflict before you merge

File: mobile/App.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import React, {useEffect} from 'react';
-import {Text} from 'react-native';
+import {Alert, Text} from 'react-native';
```
Comment (lasharela) [inline mobile/App.tsx:2]: Alert probably for debugging 

Do you still need it?

File: mobile/package.json [hunk 2]
```diff
@@ -30,7 +31,7 @@
     "metro-react-native-babel-preset": "^0.67.0",
     "react-native-svg-transformer": "^1.0.0",
     "react-test-renderer": "17.0.2",
-    "typescript": "^4.4.4"
+    "typescript": "^4.6.3"
```
Comment (lasharela) [inline mobile/package.json:34]: 4.4 was not good enough? :)

File: mobile/src/components/toggleButton/ToggleButton.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}) => {
+    const [getSelectionMode, setSelectionMode] = useState(selectionMode);
+
+    const updatedSwitchData = (id: boolean) => {
+        setSelectionMode(id);
+    }
+    return (
+        <View>
+            <View
+                style={{
```
Comment (lasharela) [inline mobile/src/components/toggleButton/ToggleButton.tsx:26]: Why this style is inline?

File: mobile/src/components/toggleButton/ToggleButton.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    switchData = false,
+    ...rest
+}) => {
+    const [getSelectionMode, setSelectionMode] = useState(selectionMode);
+
+    const updatedSwitchData = (id: boolean) => {
+        setSelectionMode(id);
+    }
+    return (
+        <View>
```
Comment (lasharela) [inline mobile/src/components/toggleButton/ToggleButton.tsx:24]: Do you really need 2 views?

File: mobile/src/components/toggleButton/ToggleButton.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                        }}
+                    >
+                        {primaryText}
+                    </Text>
+                </TouchableOpacity>
+                <TouchableOpacity
+                    onPress={()=> updatedSwitchData(false)}
+                        style={[
+                            styles.container,
+                            containerStyle,
```
Comment (lasharela) [inline mobile/src/components/toggleButton/ToggleButton.tsx:67]: Styles needs to be in separate file.

In-line styles have many disadvantages

File: mobile/src/components/toggleButton/ToggleButton.type.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+export type TToggleButtonProps = {
+    primaryText: string;
+    secondaryText: string;
+    selectionMode: boolean;
+    containerStyle?: StyleProp<ViewStyle>;
+    buttonHeight?: number;
+    buttonWidth?: number;
+    buttonTheme: string;
+    switchData: boolean;
```
Comment (lasharela) [inline mobile/src/components/toggleButton/ToggleButton.type.tsx:11]: More logical name? Kind a "tabs" or switches?

This component is more "tab" then switcher. It has contentas well

**Review by lasharela — COMMENTED** [review]
Needs a little polishing

**Pull request merged**